lib/util/fs/index.ts

```diff
@@ -46,6 +46,30 @@ export async function readLocalFile(
   }
 }
 
+// TODO: can return null #7154
+export function readLocalFileSync(fileName: string): Buffer;
+export function readLocalFileSync(fileName: string, encoding: string): string;
+export function readLocalFileSync(
+  fileName: string,
+  encoding?: string
+): string | Buffer | null {
+  const { localDir } = GlobalConfig.get();
+  const localFileName = upath.join(localDir, fileName);
+  if (localDir && !localFileName.startsWith(localDir)) {
+    logger.warn('Preventing access to file outside the base directory');
+    return null;
+  }
```

Comment:
@rarkins I think we should add this check to all `local` functions

Reply:
Yeah, let's do it in a separate PR